dashboard/lib/pd/survey_pipeline/retriever.rb

```diff
@@ -0,0 +1,59 @@
+module Pd::SurveyPipeline
+  class RetrieverBase
+    def retrive_data(*)
+      raise 'Must override in derived class'
+    end
+  end
+
+  class DailySurveyRetriever < RetrieverBase
+    attr_reader :workshop_ids, :form_ids
+
+    # Set filters for this retriever.
+    # @param workshop_ids [Array<Integer>] array of selected workshop ids
+    # @param form_ids [Array<Integer>] array of selected form ids
+    def initialize(workshop_ids: nil, form_ids: nil)
+      @workshop_ids = workshop_ids
+      @form_ids = form_ids
+    end
+
+    # Retrieve data from Pd::SurveyQuestion, Pd::WorkshopDailySurvey
+    # and Pd::WorkshopFacilitatorDailySurvey.
+    # @return [Hash{survey_questions, workshop_surveys, facilitator_surveys}]
```

Comment:
does this mean an array of hashes? or one hash with keys 'survey_questions', 'workshop_surveys', 'facilitator_surveys'?

Reply:
It's the latter. Will update it to `Hash{Symbol => Array}` as a correct [YARD style](https://yardoc.org/types.html#see_results)